src/Dependencies/Collections/ImmutableSegmentedDictionary`2+ValueBuilder.cs

```diff
@@ -0,0 +1,244 @@
+
+            IEnumerable<TValue> IReadOnlyDictionary<TKey, TValue>.Values => throw new NotSupportedException();
+
+            ICollection<TKey> IDictionary<TKey, TValue>.Keys => throw new NotSupportedException();
+
+            ICollection<TValue> IDictionary<TKey, TValue>.Values => throw new NotSupportedException();
+
+            bool ICollection<KeyValuePair<TKey, TValue>>.IsReadOnly => false;
+
+            ICollection IDictionary.Keys => throw new NotSupportedException();
+
+            ICollection IDictionary.Values => throw new NotSupportedException();
+
+            bool IDictionary.IsReadOnly => false;
+
+            bool IDictionary.IsFixedSize => false;
+
+            object ICollection.SyncRoot => throw new NotSupportedException();
+
+            bool ICollection.IsSynchronized => false;
+
+            public TValue this[TKey key]
+            {
+                get => ReadOnlyDictionary[key];
+                set => GetOrCreateMutableDictionary()[key] = value;
+            }
+
+            object? IDictionary.this[object key]
+            {
+                get => ((IDictionary)ReadOnlyDictionary)[key];
+                set => ((IDictionary)GetOrCreateMutableDictionary())[key] = value;
+            }
+
+            private SegmentedDictionary<TKey, TValue> GetOrCreateMutableDictionary()
+            {
+                return _mutableDictionary ??= new SegmentedDictionary<TKey, TValue>(_dictionary._dictionary, _dictionary.KeyComparer);
+            }
+
+            public void Add(TKey key, TValue value)
+            {
+                if (Contains(new KeyValuePair<TKey, TValue>(key, value)))
+                    return;
+
+                GetOrCreateMutableDictionary().Add(key, value);
+            }
+
+            public void Add(KeyValuePair<TKey, TValue> item)
+                => Add(item.Key, item.Value);
+
+            public void AddRange(IEnumerable<KeyValuePair<TKey, TValue>> items)
+            {
+                if (items == null)
+                    throw new ArgumentNullException(nameof(items));
+
+                foreach (var pair in items)
+                    Add(pair.Key, pair.Value);
+            }
+
+            public void Clear()
+            {
+                if (ReadOnlyDictionary.Count != 0)
+                {
+                    if (_mutableDictionary is null)
+                        _mutableDictionary = new SegmentedDictionary<TKey, TValue>(KeyComparer);
+                    else
+                        _mutableDictionary.Clear();
+                }
+            }
+
+            public bool Contains(KeyValuePair<TKey, TValue> item)
+            {
+                return TryGetValue(item.Key, out var value)
+                    && EqualityComparer<TValue>.Default.Equals(value, item.Value);
+            }
+
+            public bool ContainsKey(TKey key)
+                => ReadOnlyDictionary.ContainsKey(key);
+
+            public bool ContainsValue(TValue value)
+            {
```

Comment:
nit: might as well use an expression body as that's what the surrounding methods do

Reply:
➡️ Updated